From a0d84e8515ea979c2f048a190ea053720788e3de Mon Sep 17 00:00:00 2001
From: Harry Youd <harry@harryyoud.co.uk>
Date: Thu, 6 Sep 2018 21:23:44 +0100
Subject: [PATCH 12/13] Explicitly initialise base class in copy constructor

Fixes:
In file included from frameworks/av/media/libmedia/include/media/IMediaPlayer.h:27:0,
                 from frameworks/av/media/libmedia/include/media/IMediaPlayerService.h:29,
                 from frameworks/av/media/libmedia/include/media/IMediaDeathNotifier.h:21,
                 from hardware/qcom/display/msm8994/libhwcomposer/hwc_qclient.h:37,
                 from hardware/qcom/display/msm8994/libhwcomposer/hwc_utils.cpp:41:
frameworks/av/include/media/VolumeShaper.h: In copy constructor 'android::media::VolumeShaper::Configuration::Configuration(const
android::media::VolumeShaper::Configuration&)':
frameworks/av/include/media/VolumeShaper.h:135:18: error: base class 'class android::Parcelable' should be explicitly initialized in the copy constructor
[-Werror=extra]
         explicit Configuration(const Configuration &configuration)
                  ^
cc1plus: all warnings being treated as errors

Change-Id: I37b77d21bc625355309a634520153c2b9d6332c8
---
 include/media/VolumeShaper.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/media/VolumeShaper.h b/include/media/VolumeShaper.h
index a3aaecef6..ece3f7bf5 100644
--- a/include/media/VolumeShaper.h
+++ b/include/media/VolumeShaper.h
@@ -126,6 +126,7 @@ public:
         Configuration()
             : Interpolator<S, T>()
             , RefBase()
+            , Parcelable()
             , mType(TYPE_SCALE)
             , mId(-1)
             , mOptionFlags(OPTION_FLAG_NONE)
@@ -135,6 +136,7 @@ public:
         explicit Configuration(const Configuration &configuration)
             : Interpolator<S, T>(*static_cast<const Interpolator<S, T> *>(&configuration))
             , RefBase()
+            , Parcelable()
             , mType(configuration.mType)
             , mId(configuration.mId)
             , mOptionFlags(configuration.mOptionFlags)
-- 
2.17.1

